examples/examples/tower-load.rs

```diff
@@ -1,410 +1,379 @@
-// }
-
-fn main() {}
+//! A demo showing how filtering on values and dynamic filter reloading can be
+//! used together to help make sense of complex or noisy traces.
+//!
+//! This example runs a simple HTTP server that implements highly advanced,
+//! cloud-native "character repetition as a service", on port 3000. The server's
+//! `GET /${CHARACTER}` route will respond with a string containing that
+//! character repeated up to the requested `Content-Length`. A load generator
+//! runs in the background and constantly sends requests for various characters
+//! to be repeated.
+//!
+//! As the load generator's logs indicate, the server will sometimes return
+//! errors, including HTTP 500s! Because the logs at high load are so noisy,
+//! tracking down the root cause of the errors can be difficult, if not
+//! impossible. Since the character-repetition service is absolutely
+//! mission-critical to our organization, we have to determine what is causing
+//! these errors as soon as possible!
+//!
+//! Fortunately, an admin service running on port 3001 exposes a `PUT /filter`
+//! route that can be used to change the trace filter for the format subscriber.
+//! By dynamically changing the filter we can try to track down the cause of the
+//! error.
+//!
+//! As a hint: all spans and events from the load generator have the "gen" target
+#![deny(rust_2018_idioms)]
+
+use bytes::Bytes;
+use futures::{
+    future::{self, Ready},
+    stream::StreamExt,
+    Future,
+};
+use http::{header, Method, Request, Response, StatusCode};
+use hyper::{server::conn::AddrStream, Body, Client, Server};
+use rand::Rng;
+use std::{
+    error::Error,
+    fmt,
+    net::SocketAddr,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+use tokio::{join, time};
+use tower::{Service, ServiceBuilder, ServiceExt};
+use tracing::{self, debug, error, info, span, trace, warn, Level, Span};
+use tracing_futures::Instrument;
+use tracing_subscriber::{filter::EnvFilter, reload::Handle, FmtSubscriber};
+use tracing_tower::{request_span, request_span::make};
+
+type Err = Box<dyn Error + Send + Sync + 'static>;
+
+#[tokio::main]
+async fn main() -> Result<(), Err> {
+    let builder = FmtSubscriber::builder()
+        .with_env_filter("info,tower_load=debug")
+        .with_filter_reloading();
+    let handle = builder.reload_handle();
+
+    let _ = tracing::subscriber::set_global_default(builder.finish());
+
+    let addr = "[::1]:3000".parse::<SocketAddr>()?;
+    let admin_addr = "[::1]:3001".parse::<SocketAddr>()?;
+
+    let admin = ServiceBuilder::new().service(AdminSvc { handle });
+
+    let svc = ServiceBuilder::new()
+        .layer(make::layer::<_, Svc, _>(req_span))
+        .service(MakeSvc);
+
+    let svc = Server::bind(&addr).serve(svc);
+    let admin = Server::bind(&admin_addr).serve(admin);
+
+    let _ = join!(
+        load_gen(&addr),
+        load_gen(&addr),
+        load_gen(&addr),
+        svc,
+        admin
+    );
```

Comment:
this will await all these tasks [_concurrently_ but not in _parallel_](https://docs.rs/tokio/0.2.11/tokio/macro.join.html#runtime-characteristics). in order to get _really_ messy log output, i think it would be nice to spawn them all separately and await their `JoinHandle`s, like 
```rust
   let _ = join!(
        tokio::spawn(load_gen(&addr)),
        tokio::spawn(load_gen(&addr)),
        tokio::spawn(load_gen(&addr)),
        tokio::spawn(svc),
        tokio::spawn(admin)
    );
```
(and make sure we're using the threadpool runtime)

Reply:
also, nit: i think we should probably use `try_join!` here, and bail if it errors, instead of just throwing out the result?

Reply:
switched over!